bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```
@@ -156,26 +161,110 @@ private void updateSchedulingBlockEnabledState(final boolean enabled) {
+                final GhostAsterism.SingleTarget gsa = (GhostAsterism.SingleTarget) a;
+                if (gsa.base().isDefined()) return new Right<>(gsa.base().get());
+                else return new Left<>(gsa.target().spTarget());
+            case GhostDualTarget:
+                final GhostAsterism.DualTarget gda = (GhostAsterism.DualTarget) a;
+                if (gda.base().isDefined()) return new Right<>(gda.base().get());
+                else {
+                    final Option<Coordinates> cOpt = ImOption.fromScalaOpt(gda.basePosition(ImOption.scalaNone()));
+                    return new Right<>(cOpt.map(SPCoordinates::new).getOrElse(SPCoordinates::new));
+                }
+            case GhostTargetPlusSky:
+                final GhostAsterism.TargetPlusSky gtsa = (GhostAsterism.TargetPlusSky) a;
+                if (gtsa.base().isDefined()) return new Right<>(gtsa.base().get());
+                else return new Left<>(gtsa.target().spTarget());
+            case GhostSkyPlusTarget:
+                final GhostAsterism.SkyPlusTarget gsta = (GhostAsterism.SkyPlusTarget) a;
+                if (gsta.base().isDefined()) return new Right<>(gsta.base().get());
+                else return new Left<>(gsta.target().spTarget());
+            case GhostHighResolutionTarget:
+                final GhostAsterism.HighResolutionTarget ghta = (GhostAsterism.HighResolutionTarget) a;
+                if (ghta.base().isDefined()) return new Right<>(ghta.base().get());
+                else return new Left<>(ghta.target().spTarget());
+            case GhostHighResolutionTargetPlusSky:
+                final GhostAsterism.HighResolutionTargetPlusSky ghtsa = (GhostAsterism.HighResolutionTargetPlusSky) a;
+                if (ghtsa.base().isDefined()) return new Right<>(ghtsa.base().get());
+                else return new Left<>(ghtsa.target().spTarget());
+        }
+
+        // Make the compiler shut up.
+        return new Right<>(new SPCoordinates());
```

I prefer to throw an exception when something unpossible happens.

---

👍